data/templates/secure_erase.py

```diff
@@ -56,6 +65,209 @@
+            match = patterna.match(progress_output)
+            if match:
+                return float(match.group(1))
+        elif self.percent > 0:
+            return self.percent
+        return 0.00
+
+    def sg_format_parser(self, drive):
+        """
+        Secure erase job progress parser for sg_format tool.
+        :param drive: drive name
+        :return: a float digital of percentage
+        """
+        return self.__sg_requests_parser(drive)
+
+    def sg_sanitize_parser(self, drive):
+        """
+        Secure erase job progress parser for sg_sanitize tool.
+        :param drive: drive name
+        :return: a float digital of percentage
+        """
+        #return float('+inf')
+        return self.__sg_requests_parser(drive)
+
+    def __run(self):
+        """
+        Get secure erase progress for secure erase task.
+        """
+        parser_mapper = {
+            "hdparm": self.hdparm_parser,
+            "scrub": self.scrub_parser,
+            "sg_format": self.sg_format_parser,
+            "sg_sanitize": self.sg_sanitize_parser
+        }
+        disk_count = len(self.disk_list)
+        percentage_list = [0.0]*disk_count
+        erase_start_flags = [False]*disk_count
+        payload = {
+            "taskId": self.parameters["taskId"],
+            "progress": {}
+        }
+        counter = 0
+        total_percent = 0.00
+        if not self.parameters["address"]:
+            self.parameters["address"] = "http://172.31.128.1:9080"
+        while True:
+            for (index, value) in enumerate(self.disk_list):
+                value = value.split("/")[-1]
+                if erase_start_flags[index]:
+                    # Check if secure erase sub-progress is alive
+                    command = 'ps aux | grep {} | grep {} | sed "/grep/d" | sed "/python/d"' \
+                        .format(self.parameters["tool"], value)
+                    erase_alive = subprocess.check_output(command, shell=True)
+                    if not erase_alive:
+                        percentage_list[index] = 100
+                    else:
+                        self.percent = percentage_list[index]
+                        percentage_list[index] = parser_mapper[self.parameters["tool"]](value)
+                else:
+                    erase_start_flags[index] = os.path.exists(self.path + value + '.log')
+            total_percent = sum(percentage_list)/disk_count
+            if total_percent == float('+inf'):
+                payload["progress"]["percentage"] = "Not Available"
+            else:
+                payload["progress"]["percentage"] = str("%.2f" % total_percent) + "%"
+            counter += 1
+            payload["progress"]["description"] = "This is the {}th polling with {}s interval" \
+                .format(str(counter), str(self.interval))
+            cmd = 'curl -X POST -H "Content-Type:application/json" ' \
+                '-d \'{}\' {}/api/1.1/notification' \
```

Comment:
prefer `/api/current` rather than `api/1.1`, this is the patter for other  template files.

Of course, just as commented in on-tasks PR, the better idea is to pass the whole `http://172.31.128.1:8080/api/1.1/notification` to this script to reduce the dependency.